src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -42,23 +44,92 @@ bool Series::validate() const
         logs.warning() << "Size of series for short term storage is wrong";
         return false;
     }
+
+    auto checkVectPositive = [](std::vector<double> v) {
+        return std::count_if(v.begin(), v.end(), [](double d){ return d >= 0.0; });
+    };
+    if (!checkVectPositive(maxInjectionModulation) || !checkVectPositive(maxWithdrawalModulation))
+    {
+        logs.warning() << "Values for PMAX series should be positive";
+        return false;
+    }
+
+    auto checkVectBetween = [](std::vector<double> v) {
+        return std::count_if(v.begin(), v.end(), [](double d){ return (d >= 0.0 && d <= 1.0); });
```

Comment:
```suggestion
        return std::all_of(v.begin(), v.end(), [](double d){ return (d >= 0.0 && d <= 1.0); });
```
Pass the vector by const ref